### PR TITLE
ViewBinding DSL, AdapterDelegateLayoutContainer bugfix

### PR DESCRIPTION
**Pull request by RomanPozdeev**

### CODE REVIEW

**Comment by DominuS-RU** [comment]
@RomanPozdeev Could you provide the same functionality for DataBinding library?

**Comment by sockeqwe** [comment]
thanks for this PR. will take a look at it Wednesday

File: library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegate.java [hunk 1]
```diff
@@ -45,6 +47,25 @@
...
+     * be contiguous. Consider using id resources to uniquely identify item view types.
+     *
+     * see detail {@link AdapterDelegatesManager#getItemViewType(T items, int position)}
+     *
+     * @param items The data source of the Adapter
+     * @param position The position in the datasource
+     * @return integer value identifying the type of the view needed to represent the item at
+     *                 <code>position</code>. Type codes need not be contiguous.
+     */
+    protected int getItemViewType(T items, int position) {
```
Comment (sockeqwe) [inline library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegate.java]: what is this good for? i don't fully understand why we need this? which problem does it solve?

Comment (RomanPozdeev) [reply library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegate.java]: This is for nested recyclerviews with shared view pool.

**Comment by RomanPozdeev** [comment]
> @RomanPozdeev Great work!
> Could you provide the support for DataBinding library?
Thanks.
There is no need for a separate DSL - ViewDataBinding implements ViewBinding.

File: kotlin-dsl-layoutcontainer/src/main/java/com/hannesdorfmann/adapterdelegates4/dsl/LayoutContainerListAdapterDelegateDsl.kt [hunk 2]
```diff
@@ -115,6 +117,7 @@ internal class DslLayoutContainerListAdapterDelegate<I : T, T>(
  *
  * @since 4.1.0
  */
+@ContainerOptions(cache = CacheImplementation.SPARSE_ARRAY)
```
Comment (sockeqwe) [inline kotlin-dsl-layoutcontainer/src/main/java/com/hannesdorfmann/adapterdelegates4/dsl/LayoutContainerListAdapterDelegateDsl.kt]: 👍

File: library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegate.java (hunk 1 above)
Comment (sockeqwe) [reply library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegate.java]: Need a bit more time to think about it.

Comment (sockeqwe) [reply library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegate.java]: I would propose the following: 
Let's not introduce this right now. Let's first introduce the ViewBinding DSL and then as a next step discuss how this change can be introduced by keeping a clean API. 

Can you please remove this from this PR?

File: library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegatesManager.java [hunk 3]
```diff
@@ -236,7 +236,12 @@ public int getItemViewType(@NonNull T items, int position) {
         for (int i = 0; i < delegatesCount; i++) {
             AdapterDelegate<T> delegate = delegates.valueAt(i);
             if (delegate.isForViewType(items, position)) {
-                return delegates.keyAt(i);
+                int itemViewType = delegate.getItemViewType(items, position);
```
Comment (sockeqwe) [inline library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegatesManager.java]: Please remove this as well.

**Pull request merged**